samples/monster/CMakeLists.txt

```diff
@@ -1,9 +1,10 @@
 set(GEN_DIR "${CMAKE_CURRENT_BINARY_DIR}/generated")
-set(FBS_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
+
+# Don't use FBS_DIR to test whether flatcc_generate_sources knows to use the correct file.
 
 flatcc_generate_sources(
     NAME monster_sample
-    DEFINITION_FILES monster.fbs
+    SCHEMA_FILES monster.fbs
     ALL
     OUTPUT_DIR "${GEN_DIR}"
```

Comment:
What happens if you also drop the OUTPUT_DIR ?

Reply:
Then it defaults to `CMAKE_CURRENT_BINARY_DIR`.

Reply:
Would it make sense to change that to CMAKE_CURRENT_BINARY_DIR/generated ?

Reply:
I think the choice of `generated` is too general.
Functions of cmake also choose `CMAKE_CURRENT_BINARY_DIR` by default.
e.g. https://cmake.org/cmake/help/latest/command/configure_file.html

Reply:
The input schema files are also relative to CMAKE_CURRENT_SOURCE_DIR.

Reply:
OK

Reply:
FYI: flatcc also has an input dir option, but I'm not sure it is worthwhile adding here.

Reply:
Is that option documented in `flatcc -h`?
I have added `-I`, which is needed to grep for included files for finding dependencies.

Reply:
-I is the option for input dirs.
flatcc -h :
`  -I<inpath>                 Search path for include files (multiple allowed)`

hmm maybe not for toplevel fbs files? Though I would think that also works.

Reply:
Yes, `-I` is documented. I thought you were talking about another option.

Reply:
Yes, I'm not sure that is precise. But I don't remember. It is unlikely that the parser driver makes such a distinction.

Reply:
Maybe it depends on the order of which the input files are given?
But I think not using `-I` is better.
Just imagine `gcc` looking in include dirs for a source file when you gave a wrong source as input.

Reply:
I assume you use gcc as an example and not as a related problem since flatcc -I paths should have nothing to do with gcc -I paths.

The flatcc -I option is intended for use when you have a library collection placed in some faraway dir and you do not want that mess in your schema files.

Reply:
Yes, as an example :smile: